backend/src/main/kotlin/clarity/backend/entity/InboxEntity.kt

```diff
@@ -0,0 +1,90 @@
+        try {
+            val db = DataManager.conn()
+            val statement = db!!.createStatement()
+            val selectStatement = """
+                SELECT * FROM Inbox WHERE user_id='${userId}' AND message_read='${0}'
+            """.trimIndent()
+            val messages = mutableListOf<Notification>()
+            val result = statement.executeQuery(selectStatement)
+            while (result.next()) {
+                messages.add(
+                    Notification(
+                    result.getString("notification_id").toInt(),
+                    result.getString("user_id").toInt(),
+                    result.getString("message"),
+                    result.getString("notification_date"),
+                    result.getString("message_read").toInt()
+                ))
+            }
+            return GetUnreadResponse(StatusResponse.Success, messages)
+        } catch(e: Exception) {
+            e.printStackTrace();
+            val resultPlaceholder = mutableListOf<Notification>()
+            return GetUnreadResponse(StatusResponse.Failure, resultPlaceholder)
+        }
+    }
+
+    fun markMessage(notification: MarkMessage) : NotificationResponse {
+        val db = DataManager.conn()
+        try {
+            val statement = db!!.createStatement()
+            val insertStatement = """
+                UPDATE Inbox SET message_read='${notification.isRead}' 
+                WHERE notification_id='${notification.notificationId}'
+            """.trimIndent()
+            val result = statement.executeUpdate(insertStatement)
+            return NotificationResponse(StatusResponse.Success, "Marked message")
+        } catch(e: Exception) {
+            e.printStackTrace();
+            return NotificationResponse(StatusResponse.Failure, "Could not mark message")
+        }
+    }
+
+    fun deleteMessage(notificationId: Int) : NotificationResponse {
+        val db = DataManager.conn()
+        try {
+            val statement = db!!.createStatement()
+            val deleteStatement = """
+                DELETE FROM Inbox WHERE user_id='${notificationId}'
+            """.trimIndent()
+            val result = statement.executeUpdate(deleteStatement)
+            return NotificationResponse(StatusResponse.Success, "Deleted Message Successfully")
+        } catch(e: Exception) {
+            e.printStackTrace();
+            return NotificationResponse(StatusResponse.Failure, "Could not delete message")
+        }
+    }
+
+    fun createNotification(notification: CreateNotification) : NotificationResponse {
+        val db = DataManager.conn()
+        try {
+            val statement = db!!.createStatement()
+            val insertStatement = """
+                INSERT INTO Inbox (user_id, message, notification_date, message_read)
+                VALUES(
+                '${notification.userId}', '${notification.message}', '${notification.notificationDate}', '${0}'
```

Comment:
should these integer values be wrapped in ''? will that convert them into strings in SQL or are they still integers? The same goes for all other SQL queries you've written. could you j check your database and make sure they're stored as integers and not strings?

Reply:
good catch, whats weird is that it caused no integrity constraints in any of the queries while inserting or selecting anything. I've changed all my files to match everything though, so should be good!

Reply:
See Section 3 in https://www.sqlite.org/datatype3.html. It looks like it tries to cast it if you pass in a string value for the int column(s). So if it's guaranteed to be integer, then we don't have to worry about it.